components/experimental/src/dimension/provider/units.rs

```diff
@@ -9,10 +9,77 @@
 //!
 //! Read more about data providers: [`icu_provider`]
 
-
-use alloc::borrow::Cow;
 use icu_provider::prelude::*;
-use tinystr::UnvalidatedTinyAsciiStr;
-use zerovec::{VarZeroVec, ZeroMap};
+use zerovec::ZeroMap;
+
+// TODO: implement the units provider
+// thoughts:
+//   use the category (e.g. length, temperature, etc.) as secondary key
+//   have four maps:
+//     - one for the long width patterns (e.g. "{0} meter")
+//     - one for the short width (e.g. "{0} m"?)
+//     - one for the narrow width (e.g. "{0} m")
+//     - one for the DisplayNames (e.g. "meter")
+
+#[icu_provider::data_struct(UnitsDisplayNameV1Marker = "units/displaynames@1")]
+#[derive(Clone, PartialEq, Debug)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_experimental::dimension::provider::units),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[yoke(prove_covariance_manually)]
+pub struct UnitsDisplayNameV1<'data> {
+    // TODO: store the pattern in a SinglePattern.
+    // TODO: use `MeasureUnit` for the units key instead of strings.
+    /// Contains the long width patterns for the units.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub long: ZeroMap<'data, Count, str>,
```

Comment:
Issue: These should use `icu_pattern::SinglePlaceholderPattern`

Reply:
I have added a TODO `// TODO: store the pattern in a SinglePattern`

because there are some pattens that are only strings and there is no placeholder. therefore, I will implmeent it in the upcomming PRs.

Reply:
@sffc 

After I use it, I got the following message
```
41 |     pub narrow: ZeroMap<'data, Count, SinglePlaceholderPattern<Cow<'data, str>>>,
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `ZeroMapKV<'zf_inner>` is not implemented for `icu_pattern::Pattern<SinglePlaceholder, Cow<'zf_inner, str>>`
   |
   = help: the following other types implement trait `ZeroMapKV<'a>`:
             char
             i8
             i16
             i32
             i64
             i128
             usize
             u8
           and 67 others
note: required by a bound in `ZeroMap`
  --> ~/icu4x/utils/zerovec/src/map/map.rs:64:8
   |
61 | pub struct ZeroMap<'a, K, V>
   |            ------- required by a bound in this struct
...
64 |     V: ZeroMapKV<'a> + ?Sized,
   |        ^^^^^^^^^^^^^ required by this bound in `ZeroMap`
```

This mean, we need to implement the `ZeroMapKV<'data>` trait for the SinglePlaceholder, could you implement it ?